### ISSUE TITLE
Trivy Cache: Redis Sentinal Support

### ISSUE THREAD

**Issue by muffin87** [aquasecurity/trivy#1115]
I everyone,

first of all thank you for creating trivy I migrated over from CoreOS's Clair and I love it so far.
I would like to host the trivy DB by myself and want to make it a little more robust.
I deployed trivy on my Kubernetes cluster which works fine but here comes to problem.
I want to make it high available, so I decided to scale the Statefulset to 2 replicas.

With more than one replica I encountered problems, because the two instances don't share the same cache (at least I think that is the problem).
Thankfully I discovered that you guys support Redis as a cache backend.
I added a simple Redis deployment which seems to solve these issues.
Now I'm nearly in the same situation as before. I have a HA trivy setup but now I need to take care of a HA Redis solution.

I saw that Redis created something which is called [Sentinel](https://redis.io/topics/sentinel).
Which seems to be the best solution when hosting Redis on a Kubernetes because the Sentinel support auto failover and other features.

Long story short I have two question:

1. Is there another way to host trivy in a HA fashion?
2. Do you support / plan to support the Redis Sentinel feature?
I did a bit of reverse engineering and found out that trivy uses [fanal](https://github.com/aquasecurity/fanal/blob/main/cache/redis.go) for caching and fanal uses [this Redis client lib](https://github.com/go-redis/redis). So in theory this would be possible by using `func NewFailoverClient(failoverOpt *FailoverOptions) *Client`?
